### ISSUE TITLE
Update button keeps showing up on Nightly. Restarting the browser via update button still remains on the same browser version

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#29971]
## Description 

Update button keeps showing up on Nightly. Restarting the browser via update button still remains on the same browser version

## Steps to Reproduce

   1. Install 1.52.69 Nightly build
   2. Use it for a while so that Update button shows up
   3. Restart browser using the Update button, remains on 1.52.69 build

## Actual result:

https://user-images.githubusercontent.com/17010094/234260978-086e4e2f-ad3c-4c2c-bc08-bd7142384f0c.mov

## Expected result:
Do not show unnecessary updates

## Reproduces how often: 

Easy

## Brave version (brave://version info)

<html><head></head>
<body>

Brave | 1.52.69 Chromium: 113.0.5672.53 (Official Build) nightly (64-bit)
-- | --
Revision | `12f5dac35d12e8f4e72d7dd11df557ef93bc046f-refs/branch-heads/5672@{#703}`
OS | Windows 11 Version 22H2 (Build 23440.1000)

</body>
</html>
## Version/Channel Information:

- Can you reproduce this issue with the current release? NA
- Can you reproduce this issue with the beta channel? NA
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? Not checked/tested

## Miscellaneous Information:

cc: @mkarolin @brave/qa-team

**Comment by MadhaviSeelam** [comment]
++ I see the issue as well in 1.52.69 Chromium: 113.0.5672.53 (Official Build) nightly in Win 11 x64

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.52.71 Chromium: 113.0.5672.53 (Official Build) nightly (64-bit)
-- | --
Revision | 12f5dac35d12e8f4e72d7dd11df557ef93bc046f-refs/branch-heads/5672@{#703}
OS | Windows 11 Version 22H2 (Build 22621.1555)
```

Confirmed no Update button shown 

![image](https://user-images.githubusercontent.com/98358127/234362460-04bd64f4-4f14-49a8-bc67-e7b012a82446.png)

**Comment by GeetaSarvadnya** [comment]
The issue is fixed and working fine in `Windows 10 x64 - 1.52.72` hence closing the issue 

```
Brave | 1.52.72 Chromium: 113.0.5672.53 (Official Build) nightly (64-bit)
-- | --
Revision | 12f5dac35d12e8f4e72d7dd11df557ef93bc046f-refs/branch-heads/5672@{#703}
OS | Windows 10 Version 21H2 (Build 19044.2846)

```
